distributor/pkg/lib/events/forwarder.go

```diff
@@ -33,39 +32,34 @@ func NewForwarder(httpClient *http.Client) *Forwarder {
 	}
 }
 
-func (f *Forwarder) Start(ctx *ExecutionContext) error {
-	serverURL := fmt.Sprintf("localhost:%d", config.Global.APIProxyPort)
+func (f *Forwarder) Start(ctx *ExecutionContext) {
 	mux := http.NewServeMux()
 	mux.Handle("/health", http.HandlerFunc(api.HealthEndpointHandler))
 	mux.Handle(config.Global.EventForwardingPath, http.HandlerFunc(f.handleEvent))
 	mux.Handle(config.Global.APIProxyPath, http.HandlerFunc(f.apiProxyHandler))
+	serverURL := fmt.Sprintf("localhost:%d", config.Global.APIProxyPort)
 
 	svr := &http.Server{
 		Addr:    serverURL,
 		Handler: mux,
 	}
+
 	go func() {
+		defer ctx.Wg.Done()
```

Comment:
is this safe? Before, the `waitgroup.Done()` was occurring after `<-ctx.Done()`

Reply:
should be safe, yes. However, I encountered somethings to make graceful shutdown yet cleaner, pls take a look at the updated PR

Reply:
LGTM! Thank you :shipit: